### PR TITLE
Add CAQL cli

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
This PR adds an experimental CAQL cli tool that can be used to fetch CAQL data and explain output.

Support direct connections to IRONdb is added to the CirconusData class.

# Examples

## caql fetch
```
; ./bin/caql --config "dev" fetch --count 3 --query 'pass{ find("duration"), 0 } | window:sum(1w, skip=1d) | stats:sum()' | jq .
# 2020-02-13 19:35:22
{
  "data": [
    [
      23970,
      23970,
      23970
    ]
  ],
  "meta": [
    {
      "tags": [
        "__check_broker:noit-lxc-1",
        "__check_bundle_id:61606",
        "__check_id:71016",
        "__check_module:json",
        "__check_name:GoAPI Stats",
        "__check_target:dhaifley-lxc.dev.circonus.net",
        "__check_uuid:e312a0cb-dbe9-445d-8346-13b0ae6a3382",
        "__name:duration"
      ],
      "label": "duration|stats:sum()",
      "kind": "numeric"
    }
  ],
  "version": "DF4",
  "head": {
    "period": 60,
    "query": "pass{ find(\"duration\"), 0 } | window:sum(1w, skip=1d) | stats:sum()",
    "count": 3,
    "start": 1581622320
  }
}
# OK after 0.675s
```

## caql explain
```
; ./bin/caql --config "dev" explain --count 3 --query 'pass{ find("duration"), 0 } | window:sum(1w, skip=1d) | stats:sum()'
# 2020-02-13 19:35:57
# CAQL Explain

>  pass{ find("duration"), 0 } | window:sum(1w, skip=1d) | stats:sum()

start = 1581622320, period = 60, count = 3

 +
 |  stats:sum(,) : number => number
 |
 |  1 x 3 samples @ 60 period
 |  took 0.156505s ..................................................
 +
   +
   |  window:sum(1w,skip=1d) : number => number
   |
   |  2 x 3 samples @ 60 period
   |  took 0.156381s .................................................
   +
     +
     |  pass(,) : all => number
     |
     |  2 x 20160 samples @ 60 period
     |  took 0.139672s ............................................
     +
       +
       |  find(duration,) : none => number
       |
       |  1 x 20160 samples @ 60 period
       |  took 0.139312s ............................................
       +
       +
       |  [constant](,) : none => number
       |
       |  1 x 20160 samples @ 60 period
       |  took 0.000325s
       +
# OK after 0.613s
```

### CODE REVIEW

File: bin/caql [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+@click.group()
+@click.option("-e", '--endpoint', default=None)
+@click.option("-a", '--account', default=None)
+@click.option("-t", '--token', default=None)
+@click.option("-c", '--config', default=None)
+@click.pass_context
+def cli(ctx, endpoint, account, token, config):
+    if config:
+        with open(Path("~/.caqlrc").expanduser(), "r") as fh:
```
Comment (cody271) [inline bin/caql]: I don't see where `~/.caqlrc` is being written to?

File: bin/caql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    else:
+        ctx.obj["endpoint"] = endpoint
+        ctx.obj["account"] = account
+        ctx.obj["token"] = token
+
+@cli.command()
+@click.option("-q", '--query', required=True)
+@click.option("-s", '--start')
+@click.option("-p", '--period', type=float, default = 60)
+@click.option("-c", '--count', type=int, default = 10)
```
Comment (cody271) [inline bin/caql]: I think this makes `-c` an ambiguous option?

File: bin/caql [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        circ.caql(query, start, period, count, convert_hists = False, explain = explain),
+        sys.stdout
+    )
+
+
+@cli.command()
+@click.option("-q", '--query', required=True)
+@click.option("-s", '--start')
+@click.option("-p", '--period', type=float, default = 60)
+@click.option("-c", '--count', type=int, default = 10)
```
Comment (cody271) [inline bin/caql]: Same here.

File: bin/caql (hunk 1 above)
Comment (HeinrichHartmann) [reply bin/caql]: This file has to be created manually. Will document this once it's a little matured.

**Pull request merged**